### PR TITLE
refactor(server): system config

### PR DESCRIPTION
**Pull request by jrasm91**
* Move server config implementation to just store a single diff in system-metadata instead of a dedicated table of single key/value overrides
* Make it much easier to add new configuration properties in the future

### CODE REVIEW

File: server/src/cores/system-config.core.ts
Comment (jrasm91) [inline server/src/cores/system-config.core.ts:1]: This file

File: server/src/migrations/1715787369686-RemoveSystemConfigTable.ts
Comment (jrasm91) [inline server/src/migrations/1715787369686-RemoveSystemConfigTable.ts:1]: This file

**Review by jrasm91 — COMMENTED** [review]
This PR has a lot of noise, but the only meaningful changes are in the system-config.core.ts file and the migration.

**Review by danieldietzler — APPROVED** [review]

**Review by zackpollard — APPROVED** [review]
Haven't tested this but the changes look good, if you want to merge it go ahead, I will give it an actual test tomorrow regardless :smile:

**Pull request merged**